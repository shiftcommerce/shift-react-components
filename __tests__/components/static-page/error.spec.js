// Libraries
import React from 'react'

// Components
import StaticPageError from '../../../src/components/static-page/error'

test('renders correct child elements and classnames', () => {
  const label = 'Lorem ipsum dolor sit amet, consectetur adipiscing elit'
  const className = 'dummy-classname'

  // act
  const wrapper = mount(
    <StaticPageError className={className}>
      { label }
    </StaticPageError>
  )

  // assert
  expect(wrapper).toMatchSnapshot()
  expect(wrapper).toIncludeText(label)
  expect(wrapper.find('div').first()).toHaveClassName(className)
})
