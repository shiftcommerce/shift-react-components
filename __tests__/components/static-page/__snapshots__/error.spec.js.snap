// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correct child elements and classnames 1`] = `
<StaticPageError
  className="dummy-classname"
>
  <div
    className="dummy-classname c-static-page-error"
  >
    <StaticPageErrorBody>
      <div
        className="c-static-page-error-body"
      >
        <h1>
          Oh no, something went wrong.
        </h1>
        <h2>
          We weren't able to process your request.
        </h2>
      </div>
    </StaticPageErrorBody>
    <StaticPageErrorDetails />
    Lorem ipsum dolor sit amet, consectetur adipiscing elit
  </div>
</StaticPageError>
`;
